numpy/core/src/multiarray/flagsobject.c

```diff
@@ -275,93 +285,49 @@ arrayflags_num_get(PyArrayFlagsObject *self)
     return PyInt_FromLong(self->flags);
 }
 
-/* relies on setflags order being write, align, uic */
-static int
-arrayflags_updateifcopy_set(PyArrayFlagsObject *self, PyObject *obj)
-{
-    PyObject *res;
-
-    if (obj == NULL) {
-        PyErr_SetString(PyExc_AttributeError,
-                "Cannot delete flags updateifcopy attribute");
-        return -1;
-    }
-    if (self->arr == NULL) {
-        PyErr_SetString(PyExc_ValueError,
-                "Cannot set flags on array scalars.");
-        return -1;
-    }
-    res = PyObject_CallMethod(self->arr, "setflags", "OOO", Py_None, Py_None,
-                              (PyObject_IsTrue(obj) ? Py_True : Py_False));
-    if (res == NULL) {
-        return -1;
-    }
-    Py_DECREF(res);
-    return 0;
-}
-
-static int
-arrayflags_aligned_set(PyArrayFlagsObject *self, PyObject *obj)
-{
-    PyObject *res;
-
-    if (obj == NULL) {
-        PyErr_SetString(PyExc_AttributeError,
-                "Cannot delete flags aligned attribute");
-        return -1;
-    }
-    if (self->arr == NULL) {
-        PyErr_SetString(PyExc_ValueError,
-                "Cannot set flags on array scalars.");
-        return -1;
-    }
-    res = PyObject_CallMethod(self->arr, "setflags", "OOO", Py_None,
-                              (PyObject_IsTrue(obj) ? Py_True : Py_False),
-                              Py_None);
-    if (res == NULL) {
-        return -1;
-    }
-    Py_DECREF(res);
-    return 0;
+#define TRUE_OR_FALSE_OBJ PyObject_IsTrue(obj) ? Py_True : Py_False
```

Comment:
This doesn't handle the case when `PyObject_IsTrue` returns `NULL` and sets an exception